engine/src/player-platform.cpp

```diff
@@ -2030,8 +2033,14 @@ Boolean MCPlayer::prepare(MCStringRef options)
 	if (m_platform_player == nil || m_should_recreate || !dontuseqt)
     {
         if (m_platform_player != nil)
+		{
+			// IM-2016-05-27: [[ Bug 17697 ]] Clear native layer when releasing player
+			SetNativeView(nil);
             MCPlatformPlayerRelease(m_platform_player);
+		}
```

Comment:
While the player could in principle be retained when the native later is detached, the native later should always be cleared when the platform player is released.

On 30 May 2016 09:40:53 BST, Peter TB Brett <notifications@github.com> wrote:
>>              MCPlatformPlayerRelease(m_platform_player);
>> +		}
>
>Are the lifetimes of the `MCPlatformPlayer` instance and the native
>layer the same?  I.e. are they always created and destroyed together?
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/livecode/livecode/pull/4081/files/4d276a6e6d5a9713676ab0a38f485c1f14317e13#r65041716

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.